Build failure with "NO_AVX2 = 1" and "DYNAMIC_ARCH = 1" and "DYNAMIC_OLDER = 1"

We've discovered a build issue when compiling OpenBlas with Dynamic Arch support enabled and AVX2 set to disabled.
We've got the following error message from the compiler building dynamic.c:
```
cc -O2 -pipe  -fstack-protector-strong -fno-strict-aliasing  -O2 -DSMALL_MATRIX_OPT -DMAX_STACK_ALLOC=2048 -DEXPRECISION -fopenmp -Wall -m64 -DF_INTERFACE_GFORT -fPIC -DDYNAMIC_ARCH -DDYNAMIC_OLDER -DNO_AVX2 -DSMP_SERVER -DUSE_OPENMP -DNO_WARMUP -DMAX_CPU_NUMBER=64 -DMAX_PARALLEL_NUMBER=1 -DBUILD_SINGLE=1 -DBUILD_DOUBLE=1 -DBUILD_COMPLEX=1 -DBUILD_COMPLEX16=1 -DVERSION=\"0.3.26\" -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME -UCHAR_CNAME -DASMNAME= -DASMFNAME=_ -DNAME=_ -DCNAME= -DCHAR_NAME=\"_\" -DCHAR_CNAME=\"\" -DNO_AFFINITY -I. -O2 -DSMALL_MATRIX_OPT -DMAX_STACK_ALLOC=2048 -DEXPRECISION -fopenmp -Wall -m64 -DF_INTERFACE_GFORT -fPIC -DDYNAMIC_ARCH -DDYNAMIC_OLDER -DNO_AVX2 -DSMP_SERVER -DUSE_OPENMP -DNO_WARMUP -DMAX_CPU_NUMBER=64 -DMAX_PARALLEL_NUMBER=1 -DBUILD_SINGLE=1 -DBUILD_DOUBLE=1 -DBUILD_COMPLEX=1 -DBUILD_COMPLEX16=1 -DVERSION=\"0.3.26\" -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME -UCHAR_CNAME -DASMNAME=dynamic -DASMFNAME=dynamic_ -DNAME=dynamic_ -DCNAME=dynamic -DCHAR_NAME=\"dynamic_\" -DCHAR_CNAME=\"dynamic\" -DNO_AFFINITY -I../.. -c dynamic.c -o dynamic.o

dynamic.c:734:15: error: use of undeclared identifier 'gotoblas_SAPPHIRERAPIDS'
             return &gotoblas_SAPPHIRERAPIDS;
                     ^
dynamic.c:1054:20: error: use of undeclared identifier 'gotoblas_SAPPHIRERAPIDS'
  if (gotoblas == &gotoblas_SAPPHIRERAPIDS) return corename[26];
                   ^
2 errors generated.
```
Looking at the drivers/other/dynamic.c code it seems that there is a missing definition of the gotoblas_SAPPHIRERAPIDS to gotoblas_SANDYBRIDGE mapping around the line 277 that leads to this issue.

We were able to fix this issue adding 
```
#define gotoblas_SAPPHIRERAPIDS gotoblas_SANDYBRIDGE
```
to drivers/other/dynamic.c at line 277 but I'm not quite sure if this is a right way to fix it.

--- a/driver/others/dynamic.c
+++ b/driver/others/dynamic.c
@@ -275,6 +275,7 @@ extern gotoblas_t  gotoblas_EXCAVATOR;
 #define gotoblas_SKYLAKEX gotoblas_SANDYBRIDGE
 #define gotoblas_COOPERLAKE gotoblas_SANDYBRIDGE
 #define gotoblas_ZEN gotoblas_SANDYBRIDGE
+#define gotoblas_SAPPHIRERAPIDS gotoblas_SANDYBRIDGE
 #else
 extern gotoblas_t  gotoblas_HASWELL;
 extern gotoblas_t  gotoblas_ZEN;

